ui/src/components/microsegmentation/GraphicalServiceRow.js

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2020 Verizon Media
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import Icon from '../denali/icons/Icon';
+import { colors } from '../denali/styles';
+import styled from '@emotion/styled';
+import { css, keyframes } from '@emotion/react';
+import ReactFlowRenderer from './ReactFlowRenderer';
+import { ReactFlowProvider } from 'react-flow-renderer';
+
+const TdStyled = styled.td`
+    background-color: ${(props) => props.color};
+    text-align: ${(props) => props.align};
+    padding: 5px 0 5px 15px;
+    vertical-align: middle;
+    word-break: break-all;
+`;
+
+const colorTransition = keyframes`
+        0% {
+            background-color: rgba(21, 192, 70, 0.20);
+        }
+        100% {
+            background-color: transparent;
+        }
+`;
+
+const TrStyled = styled.tr`
+    ${(props) =>
+        props.isSuccess &&
+        css`
+            animation: ${colorTransition} 3s ease;
+        `}
+    box-sizing: border-box;
+    margin-top: 10px;
+    box-shadow: 0 1px 4px #d9d9d9;
+    border: 1px solid #fff;
+    -webkit-border-image: none;
+    border-image: none;
+    -webkit-border-image: initial;
+    border-image: initial;
+    height: 50px;
+`;
+
+const TrStyledExpanded = styled.tr`
+    box-sizing: border-box;
+    margin-top: 10px;
+    box-shadow: 0 1px 4px #d9d9d9;
+    border: 1px solid #fff;
+    -webkit-border-image: none;
+    border-image: none;
+    -webkit-border-image: initial;
+    border-image: initial;
+    height: 50px;
+`;
+
+const LeftMarginSpan = styled.span`
+    margin-right: 10px;
+    vertical-align: bottom;
+`;
+
+const StyledDiv = styled.div`
+    padding: 10px 0 10px 0;
+    width: 100%;
+    height: 600px;
+`;
+
+export default class GraphicalServiceRow extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = props.api;
+        this.toggleView = this.toggleView.bind(this);
+        this.createElementsList = this.createElementsList.bind(this);
+        this.state = {
+            name: this.props.name,
+            errorMessage: null,
+            expandedView: false,
+        };
+    }
+
+    toggleView() {
+        if (this.state.expandedView) {
```

Comment:
can be simplified to this.setState({
                expandedView: !this.state.expandedView,
            });